src/Build/BackEnd/BuildManager/BuildManager.cs

```diff
@@ -404,13 +410,57 @@ public void BeginBuild(BuildParameters parameters, IEnumerable<DeferredBuildMess
             _deferredBuildMessages = null;
         }
 
+        private void UpdatePriority(Process p, ProcessPriorityClass priority)
+        {
+            try
+            {
+                p.PriorityClass = priority;
```

Comment:
Put in node reuse logic: if should be normal priority && is low priority && not windows then nodeReuse is false (even if true); otherwise, adjust priority.